src/courses/mappers/11.md

```diff
@@ -0,0 +1,54 @@
+---
+order: 11
+next: 12.md
+title: Mapper Preparations - Setting Up
+author: Charles Hu
+---
+
+## Environment Set Up
+
+Node.js (a runtime environment for JavaScript/TypeScript) and Yarn (a package manager for JavaScript/TypeScript) are external utilities which are utilized extensively within this guide. The following section details their installation process.
+
+Linux/Mac OS:
+
+1. Install [nvm](https://github.com/nvm-sh/nvm#install--update-script).
+  
+  - 1a. Use either of the following commands to install nvm:
+    ```shell
+    curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.7/install.sh | bash
+    ```
+    ```shell
+    wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.7/install.sh | bash
+    ```
+  
+  - 1b. Either restart the terminal or run the following commands to use nvm:
+    ```shell
+    export NVM_DIR="$([ -z "${XDG_CONFIG_HOME-}" ] && printf %s "${HOME}/.nvm" || printf %s "${XDG_CONFIG_HOME}/nvm")"
+    [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh" # This loads nvm
+    ```
+
+2. Run the following command to install and use Node.js v18:
+```shell
+nvm install 18
+```
+
+3. Install [Yarn](https://yarnpkg.com/getting-started/install)
+
+Windows:
+
+1. Install [Node.js v18 via the installer](https://nodejs.org/en/download/).
+
+2. Install [Yarn](https://yarnpkg.com/getting-started/install):
+
+## Repository Set Up
+
+1. Fork/branch a development repository from the main [Heimdall2 GitHub repository](https://github.com/mitre/heimdall2).
+   
+   - SAF team developers have write access to the main repository and should [create a branch](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/making-changes-in-a-branch/managing-branches#creating-a-branch) on the primary development repository. Non-SAF team developers should instead [create a fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo#forking-a-repository) of the main repository and create a development branch there.
+
+2. Create a draft [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request#creating-the-pull-request) for your development branch against the main repository branch.
+
+3. Install the necessary dependencies for Heimdall2. Under the `heimdall2` directory, enter the following command in the terminal:
+``` shell
+yarn install
```

Comment:
should we tell people to do --frozen-lockfile so as to reduce them accidentally screwing things up?  should we mention the --registry https://registry.npmjs.org flag so that they don't screw up the lockfiles / have to deal with the weirdness we're experiencing with the yarn cdn?

Reply:
` --frozen-lockfile` is probably a good thing to mention (actually we should probably make this the default behavior).
I don't think `--registry https://registry.npmjs.org/` is necessary though.